### ISSUE TITLE
False negative for consider-using-augmented-assign for nested dicts

### ISSUE THREAD

**Issue by EugeneZelenko** [pylint-dev/pylint#8959]
### Bug description

`consider-using-augmented-assign` did not catch next situations:

```
dict_of_dicts = {
    'key1': {
        'key2': 0
    }
}

dict_of_dicts['key1']['key2'] = dict_of_dicts['key1']['key2'] + 1  # [consider-using-augmented-assign]
```

### Command used

```shell
pylint --load-plugins=pylint.extensions.code_style --enable=all pylint_test.py
```

### Pylint output

```shell
************* Module pylint_test
pylint_test.py:1:0: C0114: Missing module docstring (missing-module-docstring)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```

### Expected behavior

Warning should be shown.

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.4 (main, Jun 26 2023, 16:57:25) [GCC 7.5.0]
```

**Comment by crazybolillo** [comment]
Interesting, I think I might be able to get in a PR for this in a few days.

**Comment by EugeneZelenko** [comment]
Same is true for regular `dict`:

```
a_dict = {
    'key': 0
}

a_dict['key'] = a_dict['key'] + 1  # [consider-using-augmented-assign]
```

**Comment by crazybolillo** [comment]
It seems like it also applies for lists:
```python
kiwi = [1, 2]
kiwi[0] = kiwi[0] + 1  # [consider-using-augmented-assign]
```

**Comment by crazybolillo** [comment]
PR is ready, wonder if you guys can take a look? Thanks

**Comment by EugeneZelenko** [comment]
@crazybolillo: Thank you for quick implementation!
